File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Helper/Shooter.java
package org.firstinspires.ftc.teamcode.Helper;

import com.bylazar.configurables.annotations.Configurable;

@Configurable
public class Shooter {
    public static double intakePower = 1;
    public static double outtakeTopPower = 0.75;
    public static double outtakeMiddlePower = 1;
    public static double outtakeBottomPower = 1;
    public boolean intakeOn;
    public boolean outtakeOn;

    public Shooter() {
        SetIntake(false);
        SetOuttake(false);
    }

    public void SetIntake(boolean status) {
        intakeOn = status;
        Hardware.intake.setPower(status ? intakePower : 0.0);
    }

    public void SetOuttake(boolean status) {
        outtakeOn = status;
        Hardware.outtakeBottom.setPower(status ? outtakeBottomPower : 0.0);
        Hardware.outtakeMiddle.setPower(status ? outtakeMiddlePower : 0.0);
        Hardware.outtakeTop.setPower(status ? outtakeTopPower : 0.0);
    }

    public void increaseIntakePower(double power) {
        intakePower = intakePower + power;
        if (intakePower > 1) {
            intakePower = 1;
        }
        SetIntake(intakeOn);
    }

    public void decreaseIntakePowerMultiplier(double power) {
        intakePower = intakePower - power;
        if (intakePower < 0.05) {
            intakePower = 0.05;
        }
        SetIntake(intakeOn);
    }

    public void setOuttakeTopPowerMultiplier(double power) {
        outtakeTopPower = power;
        if (outtakeTopPower > 1) {
            outtakeTopPower = 1;
        } else if (outtakeTopPower < 0.05) {
            outtakeTopPower = 0.05;
        }
        SetOuttake(outtakeOn);
    }

    public void increaseTopOuttakePower(double power) {
        outtakeTopPower = outtakeTopPower + power;
        if (outtakeTopPower > 1) {
            outtakeTopPower = 1;
        }
        SetOuttake(outtakeOn);
    }

    public void decreaseTopOuttakePower(double power) {
        outtakeTopPower = outtakeTopPower - power;
        if (outtakeTopPower < 0.05) {
            outtakeTopPower = 0.05;
        }
        SetOuttake(outtakeOn);
    }

    public void ToggleIntake() {
        SetIntake(!intakeOn);
    }

    public void ToggleOuttake() {
        SetOuttake(!outtakeOn);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Helper/Shooter.kt
package org.firstinspires.ftc.teamcode.Helper

import com.bylazar.configurables.annotations.Configurable

@Configurable
class Shooter {
    var intake: Boolean
        get() {
            return Hardware.intake.power > 0.0
        }
        set(value) {
            Hardware.intake.power = if (value) intakePower else 0.0
        }
    var outtake: Boolean
        get() {
            return Hardware.outtake.power > 0.0
        }
        set(value) {
            Hardware.outtake.power = if (value) outtakePower else 0.0
        }

    init {
        intake = false
        outtake = false
    }

    fun setOuttakeTopPowerMultiplier(power: Double) {
        outtakePower = power
        if (outtakePower > 1) {
            outtakePower = 1.0
        } else if (outtakePower < 0.05) {
            outtakePower = 0.05
        }
        outtake = outtake
    }

    fun increaseTopOuttakePower(power: Double) {
        outtakePower = outtakePower + power
        if (outtakePower > 1) {
            outtakePower = 1.0
        }
        outtake = outtake
    }

    fun decreaseTopOuttakePower(power: Double) {
        outtakePower = outtakePower - power
        if (outtakePower < 0.05) {
            outtakePower = 0.05
        }
        outtake = outtake
    }

    fun ToggleIntake() {
        intake = !intake
    }

    fun ToggleOuttake() {
        outtake = !outtake
    }

    companion object {
        @JvmField
        var intakePower: Double = 1.0

        @JvmField
        var outtakePower: Double = 1.0
    }
}
